### PR TITLE
:book: Remove roadmap.md from the book

### LINKED ISSUE #8451: Remove link to roadmap discussion from the book
### What steps did you take and what happened?

https://cluster-api.sigs.k8s.io/roadmap.html is linked to a stale discussion in GitHub

I suggest replacing this with something like "discussion about CAPI roadmap happens during weekly office hours" (we will find the exact phrasing when working on the PR

### What did you expect to happen?

To avoid pointing users to stale documents

### Cluster API version

main, 1.4, might be even older versions

### Kubernetes version

_No response_

### Anything else you would like to add?

_No response_

### Label(s) to be applied

/kind documentation
/triage accepted
/good-first-issue

### PR DESCRIPTION
**Pull request by gantanikhilraj**
Removed link to roadmap discussion from the book and,
Replaced the text :
From,
You can find the Cluster API roadmap discussion at #5556 . Please feel free to participate!
to
The Cluster API roadmap discussion will take place during the weekly office hours.

Fixes #8451

### CODE REVIEW

File: docs/book/src/roadmap.md [hunk 1]
```diff
@@ -1,3 +1,3 @@
 # Cluster API Roadmap
 
-You can find the Cluster API roadmap discussion at [GitHub](https://github.com/kubernetes-sigs/cluster-api/discussions/5556). Please feel free to participate!
+The Cluster API roadmap discussion will take place during the weekly office hours.
```
Comment (vincepri) [inline docs/book/src/roadmap.md:3]: Should we just delete this page?

Comment (killianmuldoon) [reply docs/book/src/roadmap.md:3]: +1

**Comment by killianmuldoon** [comment]
@gantanikhilraj do you have time to follow up and remove this page + remove the link to this page from the summary?

**Comment by gantanikhilraj** [comment]
> @gantanikhilraj do you have time to follow up and remove this page + remove the link to this page from the summary?

Ok @killianmuldoon , I will do it.

**Comment by killianmuldoon** [comment]
/retitile :book: Remove roadmap.md from the book

**Pull request merged**